web/src/main/java/org/springframework/security/web/server/header/DelegatingServerHttpHeadersWriter.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.server.header;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.security.web.server.util.matcher.ServerWebExchangeMatcher;
+import org.springframework.util.Assert;
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * Delegates to a provided {@link ServerHttpHeadersWriter} when
+ * {@link ServerWebExchangeMatcher#matches(ServerWebExchange)} returns a match.
+ *
+ * @author David Herberth
+ * @since 5.7
+ */
+public final class DelegatingServerHttpHeadersWriter implements ServerHttpHeadersWriter {
```

Comment:
I think the name of the class should be `ServerWebExchangeDelegatingServerHttpHeadersWriter` to align with those already existing, like `ServerWebExchangeDelegatingServerAccessDeniedHandler` and `ServerWebExchangeDelegatingReactiveAuthenticationManagerResolver`.

Reply:
I am happy to rename this, but I tried following the other `ServerHttpHeadersWriter` implementations in the same [package](https://github.com/spring-projects/spring-security/tree/main/web/src/main/java/org/springframework/security/web/server/header), where none start with `ServerWebExchange*`, e.g. the `CompositeServerHttpHeadersWriter` which has a similar usecase to the delegating implementation.

Reply:
I see. But simply reading `DelegatingServerHttpHeadersWriter` doesn't tell me exactly what this implementation is doing, actually, it tells me that it delegates to another implementation, there is no clue that a `ServerWebExchangeMatcher` will be used.

Reply:
That makes sense!